staging/src/k8s.io/client-go/util/consistencydetector/data_consistency_detector.go

```diff
@@ -79,6 +83,22 @@ func CheckDataConsistency[T runtime.Object, U any](ctx context.Context, identity
 	}
 }
 
+// areParametersValidForListCall ensures that we can form a valid LIST requests
+// for checking data consistency.
+func areParametersValidForListCall(resourceVersion string, options metav1.ListOptions) bool {
+	// since we are setting ResourceVersionMatch to metav1.ResourceVersionMatchExact
+	// we need to make sure that the continuation hasn't been set
+	// https://github.com/kubernetes/kubernetes/blob/be4afb9ef90b19ccb6f7e595cbdb247e088b2347/staging/src/k8s.io/apimachinery/pkg/apis/meta/internalversion/validation/validation.go#L38
+	hasContinuation := len(options.Continue) > 0
+
+	// since we are setting ResourceVersionMatch to metav1.ResourceVersionMatchExact
+	// we need to make sure that the RV is valid because the validation code forbids RV == "0"
+	// https://github.com/kubernetes/kubernetes/blob/be4afb9ef90b19ccb6f7e595cbdb247e088b2347/staging/src/k8s.io/apimachinery/pkg/apis/meta/internalversion/validation/validation.go#L44
+	validResourceVersion := resourceVersion != "0"
+
```

Comment:
we could also check `options.ResourceVersionMatch`, if set to `metav1.ResourceVersionMatchExact` we could skip.

Reply:
This is an optimization, we cna get back to it later.